### ISSUE TITLE
How to costum my own optimizer ?

### ISSUE THREAD

**Issue by NawelAr** [keras-team/keras#14529]
Hello,
I'm working on comparing the performances of SVRG, SAG ... for deep learning minimization. 
How can costum optimizers be implemented with keras, I tried looking at the SGD implementation but couldn't find the code source for `tf.raw_ops.ResourceApplyGradientDescent`.

Thank you !

**Comment by saichatla** [comment]
https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer
https://mlguy.medium.com/adding-custom-loss-and-optimizer-in-keras-e255764e1b7d

These two links should help you make a custom optimizer.

**Comment by jvishnuvardhan** [comment]
@NawelAr You can also check [tf-addons](https://www.tensorflow.org/addons) repo for more optimizers and also to find how to implement your own. For example, check this code for [rectified Adam](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/rectified_adam.py) optimizer. 

This GitHub keras repo is mainly for bugs and performance related issues. For any discussion on support issues, please post them on [TF Forum](https://discuss.tensorflow.org/tags) where many users/researches discusses and learn from each other. Thanks!
